### DISCUSSION TITLE
ReferenceError: BigInt is not defined

### DISCUSSION THREAD

**Discussion by dsoftware763** [panva/jose#40] [Q&A]
node_modules/@panva/jose/lib/help/rsa_primes.js:8
const ZERO = BigInt(0)
             ^

ReferenceError: BigInt is not defined
    at Object.<anonymous> (/home/kbihm/Projects/openid-backend/node_modules/@panva/jose/lib/help/rsa_primes.js:8:14)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/kbihm/Projects/openid-backend/node_modules/@panva/jose/lib/help/key_utils.js:9:23)
    at Module._compile (module.js:652:30)

Node versions tested - 10.16.0 & 12.9.1

**Comment by panva** [comment]
> Node versions tested - 10.16.0 & 12.9.1

I have CI running with 10.13.0, 10.x, 12.0.0 and 12.x on three runtimes.

BigInt has landed in node with 10.4.0

What else can you tell about your environment?

**Comment by zuizuihao** [comment]
the same problem in node v8.16.1

**Comment by panva** [comment]
@zuizuihao node 8 is [not](https://github.com/panva/jose/blob/master/README.md#usage) a supported runtime.

**Comment by zuizuihao** [comment]
> @zuizuihao node 8 is [not](https://github.com/panva/jose/blob/master/README.md#usage) a supported runtime.

alright, I just want to use openid-client which depend on this package. is there any other solution?

**Comment by dsoftware763** [comment]
> > Node versions tested - 10.16.0 & 12.9.1
> 
> I have CI running with 10.13.0, 10.x, 12.0.0 and 12.x on three runtimes.
> 
> BigInt has landed in node with 10.4.0
> 
> What else can you tell about your environment?

I am using nvm for node versions, can that be an issue?

**Comment by panva** [comment]
> I am using nvm for node versions, can that be an issue?

It may not be set up right, run `node -v` for me and `node -e 'console.log(process.versions)'` please

**Comment by panva** [comment]
@zuizuihao use 2.x of openid-client or start thinking of upgrading to node 10 or node 12. Node 8 End-of-live is December 2019 at which point it stops getting any updates. https://github.com/nodejs/Release#release-schedule

**Comment by dsoftware763** [comment]
> node -e 'console.log(process.versions)'

{
  node: '12.9.1',
  v8: '7.6.303.29-node.15',
  uv: '1.31.0',
  zlib: '1.2.11',
  brotli: '1.0.7',
  ares: '1.15.0',
  modules: '72',
  nghttp2: '1.39.2',
  napi: '4',
  llhttp: '1.1.4',
  http_parser: '2.8.0',
  openssl: '1.1.1c',
  cldr: '35.1',
  icu: '64.2',
  tz: '2019a',
  unicode: '12.1'
}

OS: Ubuntu 18.04.03

**Comment by panva** [comment]
Then this just isn't possible and there's an issue with how you run your program, try placing `console.log(process.versions)` in your process that throws `BigInt is not defined`

**Comment by dsoftware763** [comment]
One thing I forgot to mention, it gives this error when I use openid-client npm package.

**Comment by panva** [comment]
> One thing I forgot to mention, it gives this error when I use openid-client npm package.

That doesn't matter, openid-client's CI also runs the full supported runtime matrix, see [here](https://github.com/panva/node-openid-client/commit/3f8621fce70a8eb94bc48ca3170219b08e81d127/checks)

**Comment by dsoftware763** [comment]
I understand that and I agree there's something wrong with my env but that's what I am trying to figure out for the last few hours, what is going wrong, I ran the same code on another system with windows 10 OS and it ran as expected.

Just see this screenshot
![Screenshot from 2019-08-30 12-34-15](https://user-images.githubusercontent.com/48565350/64000217-e3a3cc00-cb22-11e9-8ceb-179fdbeff59d.png)

**Comment by panva** [comment]
I can see two different ways you invoke node in that screenshot, run `which nodejs` (runs in your VS), and `which node` (runs in your bash).

Its clear now your VS Code is not using your nvm installed node version and you can confirm that by placing `console.log(process.versions)` in your code that runs.
